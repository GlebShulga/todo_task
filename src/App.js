import React, { useEffect, useState } from "react";
import axios from "axios";
import CategoryList from "./components/CategoryList";
import TaskList from "./components/TaskList";
import CreateCategory from "./components/CreateCategory";
import CreateTask from "./components/CreateTask";
import "./assets/scss/App.scss";

function App() {
  const [choosenCategory, setChoosenCategory] = useState("");
  const [isNewTaskCreated, setIsNewTaskCreated] = useState(false);
  const [categoryList, setCategoryList] = useState([]);

  const fetchCategoryList = async () => {
    await axios("/api/v1/category")
      .then((res) => {
        const data = res.data;
        setCategoryList(data);
      })
      .catch((err) => console.log(err));
  };

  useEffect(() => {
    fetchCategoryList();
  }, []);

  const categoryTitleList = categoryList.map(
    (category) => category.categoryTitle
  );

  return (
    <div className="App">
      <div className="App-lists">
        <CreateCategory
          categoryTitleList={categoryTitleList}
          choosenCategory={choosenCategory}
          fetchCategoryList={fetchCategoryList}
        />
        <CreateTask
          choosenCategory={choosenCategory}
          setIsNewTaskCreated={setIsNewTaskCreated}
        />
      </div>
      <div className="App-lists">
        <CategoryList
          categoryTitleList={categoryTitleList}
          setChoosenCategory={setChoosenCategory}
          choosenCategory={choosenCategory}
          categoryList={categoryList}
          fetchCategoryList={fetchCategoryList}
        />
        <TaskList
          choosenCategory={choosenCategory}
          isNewTaskCreated={isNewTaskCreated}
        />
      </div>
    </div>
  );
}

export default App;
