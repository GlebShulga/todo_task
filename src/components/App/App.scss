.app {
  display: flex;
  flex-direction: column;
}

.app-lists {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.app-lists_padding {
  padding: 1rem;
}

.app-list {
  display: flex;
  flex-direction: column;
  margin: 0 1rem;
  width: 100%;
}

.app-header {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
}
