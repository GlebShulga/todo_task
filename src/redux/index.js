import { createStore, applyMiddleware } from "redux";
import { routerMiddleware } from "connected-react-router";
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";
import rootReducer from "./reducers";
import createHistory from "./history";

export const history = createHistory();

const initialState = {};
const enhancers = [];
const middleware = [thunk, routerMiddleware(history)];


const composedEnhancers = composeWithDevTools(
  applyMiddleware(...middleware),
  ...enhancers
);

const store = createStore(
  rootReducer(history),
  initialState,
  composedEnhancers
);

export default store;
