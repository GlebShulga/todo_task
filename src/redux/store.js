import { configureStore } from "@reduxjs/toolkit";
import category from "./reducers/category";
import task from "./reducers/task"

export const store = configureStore({
  reducer: {
    category,
    task
  },
});
