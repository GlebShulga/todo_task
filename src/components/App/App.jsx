import React, { useEffect } from "react";
import { useDispatch } from "react-redux";
import CreateCategory from "../CreateCategory/CreateCategory";
import CreateTask from "../CreateTask/CreateTask";
import ProgressBar from "../ProgressBar/ProgressBar";
import Header from "../Header/Header";
import Content from "../Content/Content";
import { fetchCategoryList } from "../../redux/actions/category";
import { fetchTaskList } from "../../redux/actions/task";
import "./App.scss";

function App() {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchCategoryList());
    dispatch(fetchTaskList());
  }, []);

  return (
    <div className="app">
      <Header />
      <ProgressBar />
      <div className="app-lists app-lists_padding">
        <CreateCategory />
        <CreateTask />
      </div>
      <Content />
    </div>
  );
}

export default React.memo(App);
