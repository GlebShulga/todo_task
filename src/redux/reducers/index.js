import { combineReducers } from 'redux'
import { connectRouter } from "connected-react-router";
import category from "./category";
import task from "./task"

const createRootReducer = (history) =>
  combineReducers({
    router: connectRouter(history),
    category,
    task,
  });

export default createRootReducer;