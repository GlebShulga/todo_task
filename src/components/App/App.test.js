import React from 'react';
import ReactDOM from "react-dom";
import renderer from 'react-test-renderer'
import { BrowserRouter as Router, MemoryRouter } from "react-router-dom";
import { Provider } from 'react-redux';
import { history } from "../../redux";
import store from '../../redux';
import App from './App';

// test('renders learn react link', () => {
//   const { getByText } = renderer.create(
//     <Provider store={store}>
//       <App />
//      </Provider>
//   );

//   expect(getByText(/To-Do List/i)).toBeInTheDocument();
// });

const mockPush = jest.fn();

jest.mock("react-router-dom", () => ({
  useLocation: jest.fn().mockReturnValue({
    pathname: "/",
    search: "",
    hash: "",
    state: null,
    key: "5nvxpbdafa",
  }),
  useHistory: () => {
    const push = () => mockPush();
    return { push };
  },
}));

describe('App', () => {
  it('renders', () => {
    const component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </Provider>
    );
      let tree = component.toJSON()
      expect(tree).toMatchSnapshot()
  })
})

