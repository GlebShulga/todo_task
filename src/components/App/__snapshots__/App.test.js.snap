// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component App snapshot 1`] = `
<div
  className="app"
>
  <div
    className="header"
    data-test="headerComponent"
  >
    <button
      className="header-title"
      data-testid="HeaderTitle"
      onClick={[Function]}
    >
      To-Do List
    </button>
    <button
      className="header-task-table_button"
      data-testid="LinkToTaskTablePageButton"
      onClick={[Function]}
    >
      Common task list
    </button>
    <div
      className="header-filter_form"
    >
      <div
        className="status"
      >
        <input
          checked={false}
          data-testid="StatusDoneCheckBox"
          id="statusDone"
          name="statusDone"
          onChange={[Function]}
          type="checkbox"
          value={true}
        />
        <label
          htmlFor="statusDone"
        >
          Show done
        </label>
      </div>
      <div
        className="header-form"
      >
        <input
          className="header-form_input"
          data-testid="HeaderFormInput"
          name="search"
          onChange={[Function]}
          placeholder="Search"
          type="text"
          value=""
        />
        <button
          className="header-form_button"
          data-testid="SearchButton"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-times fa-w-11 "
            data-icon="times"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 352 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
  <progress
    className="progress-bar"
    data-testid="ProgressBar"
    max="100"
    value={100}
  />
  <div
    className="app-lists app-lists_padding"
  >
    <div
      className="create_category"
    >
      <div
        className="create_category-form"
      >
        <input
          className="create_category-form_input"
          data-testid="AddCategoryInputField"
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder="Add new category"
          type="text"
          value=""
        />
        <button
          className="form_button_add"
          data-testid="AddCategoryButton"
          onClick={[Function]}
          type="button"
        >
          Add
        </button>
      </div>
    </div>
    <div
      className="create_task"
    >
      <div
        className="create_task-form"
      >
        <input
          className="create_task-form_input"
          data-testid="AddTaskInputField"
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder="Add new task"
          type="text"
          value=""
        />
        <button
          className="form_button_add"
          data-testid="AddTaskButton"
          onClick={[Function]}
          type="button"
        >
          Add
        </button>
      </div>
    </div>
  </div>
  <div
    className="app-lists"
  >
    <div
      className="category-list_editing-mode"
    >
      <ul
        className="category-list category-list_width scroll"
      >
        <li
          className="category-table"
        >
          <div
            className="category"
            onClick={[Function]}
            style={
              Object {
                "marginLeft": "0rem",
              }
            }
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-check-square fa-w-14 check-mark--hidden"
              data-icon="check-square"
              data-prefix="fas"
              data-testid="faCheckSquare"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M400 480H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48v352c0 26.51-21.49 48-48 48zm-204.686-98.059l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.248-16.379-6.249-22.628 0L184 302.745l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.25 16.379 6.25 22.628.001z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            <button
              className="category-title"
              data-testid="ChooseCategoryButton"
              onClick={[Function]}
            >
              test 7
            </button>
            <button
              data-testid="RenameCategoryButton"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-edit fa-w-18 "
                data-icon="edit"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </button>
            <div
              className="icon"
            >
              <button
                data-testid="DelCategoryButton"
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-trash-alt fa-w-14 "
                  data-icon="trash-alt"
                  data-prefix="fas"
                  data-testid="faTrashAlt"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </button>
              <button
                className="plus-icon"
                data-testid="CreateTaskModalButton"
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-plus fa-w-14 "
                  data-icon="plus"
                  data-prefix="fas"
                  data-testid="faPlus"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </button>
            </div>
          </div>
          <ul
            className="category-list"
          />
        </li>
        <li
          className="category-table"
        >
          <div
            className="category"
            onClick={[Function]}
            style={
              Object {
                "marginLeft": "0rem",
              }
            }
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-check-square fa-w-14 check-mark--hidden"
              data-icon="check-square"
              data-prefix="fas"
              data-testid="faCheckSquare"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M400 480H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48v352c0 26.51-21.49 48-48 48zm-204.686-98.059l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.248-16.379-6.249-22.628 0L184 302.745l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.25 16.379 6.25 22.628.001z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            <button
              className="category-title"
              data-testid="ChooseCategoryButton"
              onClick={[Function]}
            >
              category 4
            </button>
            <button
              data-testid="RenameCategoryButton"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-edit fa-w-18 "
                data-icon="edit"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </button>
            <div
              className="icon"
            >
              <button
                data-testid="DelCategoryButton"
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-trash-alt fa-w-14 "
                  data-icon="trash-alt"
                  data-prefix="fas"
                  data-testid="faTrashAlt"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </button>
              <button
                className="plus-icon"
                data-testid="CreateTaskModalButton"
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-plus fa-w-14 "
                  data-icon="plus"
                  data-prefix="fas"
                  data-testid="faPlus"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </button>
            </div>
          </div>
          <ul
            className="category-list"
          />
        </li>
        <li
          className="category-table"
        >
          <div
            className="category"
            onClick={[Function]}
            style={
              Object {
                "marginLeft": "0rem",
              }
            }
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-check-square fa-w-14 check-mark--hidden"
              data-icon="check-square"
              data-prefix="fas"
              data-testid="faCheckSquare"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M400 480H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48v352c0 26.51-21.49 48-48 48zm-204.686-98.059l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.248-16.379-6.249-22.628 0L184 302.745l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.25 16.379 6.25 22.628.001z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            <button
              className="category-title"
              data-testid="ChooseCategoryButton"
              onClick={[Function]}
            >
              category 2
            </button>
            <button
              data-testid="RenameCategoryButton"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-edit fa-w-18 "
                data-icon="edit"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </button>
            <div
              className="icon"
            >
              <button
                data-testid="DelCategoryButton"
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-trash-alt fa-w-14 "
                  data-icon="trash-alt"
                  data-prefix="fas"
                  data-testid="faTrashAlt"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </button>
              <button
                className="plus-icon"
                data-testid="CreateTaskModalButton"
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-plus fa-w-14 "
                  data-icon="plus"
                  data-prefix="fas"
                  data-testid="faPlus"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </button>
            </div>
          </div>
          <ul
            className="category-list"
          />
        </li>
        <li
          className="category-table"
        >
          <div
            className="category"
            onClick={[Function]}
            style={
              Object {
                "marginLeft": "0rem",
              }
            }
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-check-square fa-w-14 check-mark--hidden"
              data-icon="check-square"
              data-prefix="fas"
              data-testid="faCheckSquare"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M400 480H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48v352c0 26.51-21.49 48-48 48zm-204.686-98.059l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.248-16.379-6.249-22.628 0L184 302.745l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.25 16.379 6.25 22.628.001z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            <button
              className="category-title"
              data-testid="ChooseCategoryButton"
              onClick={[Function]}
            >
              category 1
            </button>
            <button
              data-testid="RenameCategoryButton"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-edit fa-w-18 "
                data-icon="edit"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </button>
            <div
              className="icon"
            >
              <button
                data-testid="DelCategoryButton"
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-trash-alt fa-w-14 "
                  data-icon="trash-alt"
                  data-prefix="fas"
                  data-testid="faTrashAlt"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </button>
              <button
                className="plus-icon"
                data-testid="CreateTaskModalButton"
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-plus fa-w-14 "
                  data-icon="plus"
                  data-prefix="fas"
                  data-testid="faPlus"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </button>
            </div>
          </div>
          <ul
            className="category-list"
          />
        </li>
        <li
          className="category-table"
        >
          <div
            className="category"
            onClick={[Function]}
            style={
              Object {
                "marginLeft": "0rem",
              }
            }
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-check-square fa-w-14 check-mark--hidden"
              data-icon="check-square"
              data-prefix="fas"
              data-testid="faCheckSquare"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M400 480H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48v352c0 26.51-21.49 48-48 48zm-204.686-98.059l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.248-16.379-6.249-22.628 0L184 302.745l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.25 16.379 6.25 22.628.001z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            <button
              className="category-title"
              data-testid="ChooseCategoryButton"
              onClick={[Function]}
            >
              category 3
            </button>
            <button
              data-testid="RenameCategoryButton"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-edit fa-w-18 "
                data-icon="edit"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </button>
            <div
              className="icon"
            >
              <button
                data-testid="DelCategoryButton"
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-trash-alt fa-w-14 "
                  data-icon="trash-alt"
                  data-prefix="fas"
                  data-testid="faTrashAlt"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </button>
              <button
                className="plus-icon"
                data-testid="CreateTaskModalButton"
                onClick={[Function]}
                type="button"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-plus fa-w-14 "
                  data-icon="plus"
                  data-prefix="fas"
                  data-testid="faPlus"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </button>
            </div>
          </div>
          <ul
            className="category-list"
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;
